libsolidity/formal/BMC.h

```diff
@@ -184,7 +182,7 @@ class BMC: public SMTEncoder
 	/// Assertions that are known to be safe.
 	std::set<Expression const*> m_safeAssertions;
 
-	std::shared_ptr<smt::SolverInterface> m_interface;
+	std::unique_ptr<smt::SolverInterface> m_interface;
```

Comment:
I'm wondering about the order of destruction here - it doesn't seem to cause problems right now, but maybe it's generally better to declare this as the *first* member, so that it is destructed last? If e.g. something in ``m_verificationTargets`` were to depend on the interface being alive, this would crash on destruction... but yeah - since apparently it's not causing problems, maybe no need.

Reply:
done